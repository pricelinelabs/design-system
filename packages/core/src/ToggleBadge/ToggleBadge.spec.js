import React from 'react'
import { ToggleBadge, theme } from '..'

describe('ToggleBadge', () => {
  let consoleError
  beforeEach(() => {
    consoleError = console.error
    console.error = jest.fn()
  })
  afterEach(() => (console.error = consoleError))

  test('selected ToggleBadge renders with default props', () => {
    const json = rendererCreateWithTheme(<ToggleBadge selected />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightBlue)
    expect(json).toHaveStyleRule('color', theme.colors.blue)
  })

  test('unselected ToggleBadge renders with default props', () => {
    const json = rendererCreateWithTheme(<ToggleBadge />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('color', theme.colors.blue)
  })
})
