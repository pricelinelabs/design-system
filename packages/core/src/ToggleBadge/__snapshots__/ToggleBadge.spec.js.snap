// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleBadge selected ToggleBadge renders with default props 1`] = `
.c0 {
  border-radius: 2px;
  border: 0;
  display: inline-block;
  font-weight: 700;
  font-family: inherit;
  cursor: pointer;
  background-color: #e8f2ff;
  color: #0068EF;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  margin-left: 4px;
  margin-right: 4px;
  margin-top: 4px;
  margin-bottom: 4px;
  font-size: 12px;
}

.c0:hover {
  background-color: #e8f2ff;
}

<button
  className="c0"
  color="primary"
  fontSize={0}
  selected={true}
/>
`;

exports[`ToggleBadge unselected ToggleBadge renders with default props 1`] = `
.c0 {
  border-radius: 2px;
  border: 0;
  display: inline-block;
  font-weight: 700;
  font-family: inherit;
  cursor: pointer;
  background-color: transparent;
  color: #0068EF;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  margin-left: 4px;
  margin-right: 4px;
  margin-top: 4px;
  margin-bottom: 4px;
  font-size: 12px;
}

.c0:hover {
  background-color: #e8f2ff;
}

<button
  className="c0"
  color="primary"
  fontSize={0}
  selected={false}
/>
`;
